### DISCUSSION TITLE
muteTime Date 객체 저장시간이 +9인 점에 대해서

### DISCUSSION THREAD

**Discussion by khk811** [42MVP/ft_transcendence#5] [General]
default muteTime을 분명

`private defaultMuteTime: Date = new Date('1970-01-01T00:00:00.000Z');`

이렇게 선언했는데 DB에는 9시간 후인 `1970-01-01 09:00:00` 이런 값으로 저장이 되는 점을 발견했습니다.

그래서 실험삼아 온라인 콘솔에서 돌려봤는데 아무래도 저 기본값은 UTC 기준인것 같습니다.
```
let sample = new Date('1970-01-01T00:00:00.000Z');
console.log(sample);

console.log(`${sample.toUTCString()}`);
console.log(`${sample.toLocaleString()}`);
```

위의 코드에서 .toUTCString()은 입력된 값과 같은 0시가 맞지만,
.toLocaleString()은 한국 시간 기준으로 (9시간이 더해져서) 출력이 되는것 같습니다. 

```
[LOG]: Date: "1970-01-01T00:00:00.000Z"  // sample
[LOG]: "Thu, 01 Jan 1970 00:00:00 GMT"   // .toUTCString()
[LOG]: "1970. 1. 1. 오전 9:00:00"         // .toLocaleString()
```
모든 Date 객체의 시간이 DB에 균일하게 9시간씩 더해져서 저장되게 된다면 문제가 되지 않을것 같습니다.

혹시 클라이언트가 요청한 muteTime 시간이 원치 않는 시간으로 변경이 되는 경우가 있을까요?

**Comment by chaejin-jen** [comment]
쓰다가 문제 있으면 말씀 드릴게요.
그런데 디폴트값으로 원한건 new Date(0); 이었습니다.
확인 후 수정 부탁드려요 !

**Reply by khk811** [reply]
new Date(0)로 수정해도 동일하게 9시간 뒤 값이 들어갑니다
<img width="701" alt="스크린샷 2023-08-07 13 55 24" src="https://github.com/42MVP/ft_transcendence/assets/26704946/48b7592d-c69a-40f7-8763-975384de11ac">

**Reply by ittzggd** [reply]
잘 모르겠지만 db timezone이  utc가 아니라  로컬(서울)로 되어있는거 아닐까용...?
